api/src/main/java/org/openmrs/module/fgh/mpi/MpiHttpClient.java

```diff
@@ -122,19 +168,110 @@ public void submitPatient(String patientData) throws Exception {
 			log.debug("Patient data -> " + patientData);
 		}
 		
-		List<Map<String, Object>> mpiIdsResp = submitRequest(SUBPATH_PATIENT, patientData, List.class);
+		AdministrationService adminService = Context.getAdministrationService();
 		
-		if (log.isDebugEnabled()) {
-			log.debug("MPI patient submission response: " + mpiIdsResp);
-		}
+		String mpiSystem = adminService.getGlobalProperty(GP_MPI_SYSTEM);
```

Comment:
I thought you already checked the system somewhere earlier, you need to cache it in a static to avoid multiple lookups

Reply:
Ok